pygeoapi/openapi.py

```diff
@@ -150,32 +150,54 @@ def gen_contact(cfg: dict) -> dict:
         'email': cfg['metadata']['contact']['email']
     }
 
-    contact['x-ogc-serviceContact'] = {
-        'name': cfg['metadata']['contact']['name'],
-        'position': cfg['metadata']['contact']['position'],
-        'addresses': [{
-            'deliveryPoint': [cfg['metadata']['contact']['address']],
-            'city': cfg['metadata']['contact']['city'],
-            'administrativeArea': cfg['metadata']['contact']['stateorprovince'],  # noqa
-            'postalCode': cfg['metadata']['contact']['postalcode'],
-            'country': cfg['metadata']['contact']['country']
-        }],
-        'phones': [{
-            'type': 'main', 'value': cfg['metadata']['contact']['phone']
-            }, {
-            'type': 'fax', 'value': cfg['metadata']['contact']['fax']
-        }],
-        'emails': [{
-            'value': cfg['metadata']['contact']['email']
-        }],
-        'contactInstructions': cfg['metadata']['contact']['instructions'],
-        'links': [{
+    service_contact = {}
+    if cfg['metadata']['contact'].get('name'):
+        service_contact['name'] = cfg['metadata']['contact']['name']
```

Comment:
Note that the `if` here checks if the attribute exists AND evaluates as `True`. This means that if a field has value `0`, it won't be copied into the output dictionary. Granted, it's unlikely to be the case here, but I'd suggest to use `if "name" in cfg['metadata']['contact']`.
```python
def dmap(source: dict, mapping: dict, target:dict = None) -> dict:
    """Map keys from source to target using mapping."""
    if target is None:
        target = {}
    else:
        target = target.copy()

    for key, val in mapping.items():
        if key in source:
            target[val] = source[key]

    return target
```

There's apparently a complete library to make those types of conversions: https://pypi.org/project/json-converter/